mollie/refunds.go

```diff
@@ -0,0 +1,243 @@
+type EmbedValue string
+
+// Valid Embed query string value.
+const (
+	EmbedPayment EmbedValue = "payment"
+)
+
+// ListRefundOptions describes list refund endpoint valid query string parameters.
+//
+// See: https://docs.mollie.com/reference/v2/refunds-api/list-refunds.
+type ListRefundOptions struct {
+	From      string     `url:"from,omitempty"`
+	Limit     string     `url:"limit,omitempty"`
+	ProfileID string     `url:"profileId,omitempty"`
+	TestMode  bool       `url:"testmode,omitempty"`
+	Embed     EmbedValue `url:"embed,omitempty"`
+}
+
+// RefundsService instance operates over refund resources.
+type RefundsService service
+
+// Get retrieve a single refund by its ID.
+//
+// If you do not know the original payment’s ID, you can use the List payment refunds endpoint.
+func (rs *RefundsService) Get(paymentID, refundID string, options *RefundOptions) (refund Refund, err error) {
+	u := fmt.Sprintf("v2/payments/%s/refunds/%s", paymentID, refundID)
+	if options != nil {
+		v, _ := query.Values(options)
+		u = fmt.Sprintf("%s?%s", u, v.Encode())
+	}
+
+	req, err := rs.client.NewAPIRequest(http.MethodGet, u, options)
+	if err != nil {
+		return
+	}
+
+	res, err := rs.client.Do(req)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(res.content, &refund); err != nil {
+		return
+	}
+
+	return
+}
+
+var (
+	requiredCreateParamRefund = "parameter required for creating a refund: %+v"
+)
+
+// Create a refund payment request.
+//
+// See https://docs.mollie.com/reference/v2/refunds-api/create-refund.
+func (rs *RefundsService) Create(paymentID string, re Refund, options *RefundOptions) (rf Refund, err error) {
+	if re.Amount == nil {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount)
+	}
+
+	if re.Amount.Currency == "" {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount.Currency)
+	}
+
+	if re.Amount.Value == "" {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount.Value)
+	}
+
+	u := fmt.Sprintf("v2/payments/%s/refunds", paymentID)
+	if options != nil {
+		v, _ := query.Values(options)
+		u = fmt.Sprintf("%s?%s", u, v.Encode())
+	}
+
+	req, err := rs.client.NewAPIRequest(http.MethodPost, u, re)
+	if err != nil {
+		return
+	}
+
+	res, err := rs.client.Do(req)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(res.content, &rf); err != nil {
+		return
+	}
+
+	return
+}
+
+// Cancel try to cancel the refund request.
+// The refund can only be canceled while the refund’s status is either queued or pending.
+// See https://docs.mollie.com/reference/v2/refunds-api/cancel-refund
+func (rs *RefundsService) Cancel(paymentID, refundID string, options *RefundOptions) (err error) {
+	u := fmt.Sprintf("v2/payments/%s/refunds/%s", paymentID, refundID)
+	if options != nil {
+		v, _ := query.Values(options)
+		u = fmt.Sprintf("%s?%s", u, v.Encode())
+	}
+
+	req, err := rs.client.NewAPIRequest(http.MethodDelete, u, nil)
+	if err != nil {
+		return
+	}
+
+	_, err = rs.client.Do(req)
```

Comment:
Why are you ignoring the response?

Remember, this is a client aimed for people to build things on top off, so even when it might seem like the response if not relevant we should not take that desition for them.

Please process and return the response.

Reply:
Ok. I'll fix it.

Reply:
But what I should return if the response from Mollie is `204 No Content` ?
As documented [here](https://docs.mollie.com/reference/v2/refunds-api/cancel-refund)

Reply:
I just stumbled upon the same error, in that case the function should only return an error. I did something like that in #33

Reply:
Great. It's the same as my code. Now I can mark this as resolved.

Reply:
@VictorAvelar  I know it's already too late to discuss this, but it keeps bugging me. You said:

> 
> 
> I just stumbled upon the same error, in that case the function should only return an error. I did something like that in #33

You also said:

> 
> 
> Why are you ignoring the response?
> 
> Remember, this is a client aimed for people to build things on top off, so even when it might seem like the response if not relevant we should not take that desition for them.
> 
> Please process and return the response.

It's a little bit contradictory for me. Shouldn't we return the response header also, rather than just returning the object and error?
Something like this:
`func (rs *RefundsService) Get(paymentID, refundID string, options *RefundOptions) (resHeader http.Response, refund Refund, err error) {`
What do you think about it?

Sorry to bother you. I think my commits looks messy, just now I learn about squashing. I'm new to git also :sweat_smile:

Reply:
Hey @dwikur21 yes it is contradictory, I will put it in the right order and hopefully that will help to clarify the topic.

1. I asked you to not ignore the response and so on.
2. While working on the profiles API implementation I stumbled upon the same situation for the 204, which led me to the solution implemented in #33 
3. I'm currently doing a bit of research to see how other libraries are handling these cases, so probably I will open a task whenever I find a pattern that resolves this type of conflicts.
For what I can see so far they are returning 3 values for all the operations, I will use the Get refund method as an example:
```go
func (rs *RefundsService) Get(pID string, rID string, options *RefundOptions) (refund *Refund, response *http.Response, err error) {
    // method body
}
```
So what they do is returning `object, response, error`, for me it kinda makes sense but I still want to dig in a bit more. 

p.s. You don't need to be sorry, coding is a continuous learning process for everybody, you're doing great.

Reply:
Great that you are already thinking about it. Thanks @VictorAvelar :smile: